lib/mpool/src/mpool.c

```diff
@@ -562,6 +586,8 @@ mpool_cparams_defaults(struct mpool_cparams *cparams)
 
     strlcpy(cparams->mclass[MP_MED_CAPACITY].path, MPOOL_CAPACITY_MCLASS_DEFAULT_PATH,
             sizeof(cparams->mclass[MP_MED_CAPACITY].path));
+
+    cparams->pmem_only = false;
```

Comment:
This feels a little bit like abuse to me. Would rather pmem-only not get tacked onto the cparams struct unless it can be set with paramc/paramv.

Reply:
"pmem_only" is not exposed to an user via kvdb cparams struct. This is evaluated internally by a kvdb by running few checks on the KVDB home and media class paths, and passed as a mpool create/run time parameter. I think this is fine, but if you strongly insist I can add this as a parameter to mpool_create() API.

Reply:
Can you push all the dax/pmem_only logic into mpool_create/open?  Then provide an API to ask mpool if it is pmem only.

Reply:
Please check - https://github.com/hse-project/hse/pull/97#discussion_r750834244